nilearn/input_data/tests/test_nifti_masker.py

```diff
@@ -228,27 +228,29 @@ def test_joblib_cache():
     mask[20, 20, 20] = 1
     mask_img = Nifti1Image(mask, np.eye(4))
 
-    with testing.write_tmp_imgs(mask_img, create_files=True)\
-            as filename:
+    with testing.write_tmp_imgs(mask_img, create_files=True) as filename:
         masker = NiftiMasker(mask_img=filename)
         masker.fit()
         mask_hash = hash(masker.mask_img_)
         masker.mask_img_.get_data()
         assert_true(mask_hash == hash(masker.mask_img_))
 
-    # Test a tricky issue with memmapped joblib.memory that makes
-    # imgs return by inverse_transform impossible to save
-    cachedir = mkdtemp()
-    try:
-        masker.memory = Memory(cachedir=cachedir, mmap_mode='r',
-                               verbose=0)
-        X = masker.transform(mask_img)
-        # inverse_transform a first time, so that the result is cached
-        out_img = masker.inverse_transform(X)
-        out_img = masker.inverse_transform(X)
-        out_img.to_filename(os.path.join(cachedir, 'test.nii'))
-    finally:
-        shutil.rmtree(cachedir, ignore_errors=True)
+        # Test a tricky issue with memmapped joblib.memory that makes
+        # imgs return by inverse_transform impossible to save
+        cachedir = mkdtemp()
+        try:
+            masker.memory = Memory(cachedir=cachedir, mmap_mode='r',
+                                   verbose=0)
+            X = masker.transform(mask_img)
+            # inverse_transform a first time, so that the result is cached
+            out_img = masker.inverse_transform(X)
+            out_img = masker.inverse_transform(X)
+            out_img.to_filename(os.path.join(cachedir, 'test.nii'))
+            # ensure the file handler contained in masker is released by
+            # deleting it, this avoids an error on windows
+            del masker
```

Comment:
Should the del be in the finally: if you get an error here, you won't del the masker, and thus the finally will fail.

The way I would write this would be:

* before the 'try': masker = None (so that the del is valid even if instanciation of the masker failed).
* in the 'finally': del masker